CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+if (NOT DEFINED FREERTOS_MEMMANG)
+    message(FATAL_ERROR "FREERTOS_MEMMANG not defined")
+endif()
+
+if (NOT DEFINED FREERTOS_PORT)
+    message(FATAL_ERROR "FREERTOS_PORT not defined")
+endif()
+
+set(
+    FREERTOS_INC
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+)
+
+set(
+    FREERTOS_SRC
+    ${CMAKE_CURRENT_SOURCE_DIR}/tasks.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/timers.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/stream_buffer.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/queue.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/list.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/event_groups.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/croutine.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/portable/MemMang/${FREERTOS_MEMMANG}.c
```

Comment:
In the interest of accommodating all common use cases (in addition to laying the foundation for using Cmake on other targets), you might want to consider allowing FREERTOS_MEMMANAGE to accommodate a user defined solution. The most common ones used on STM and NXP ARM systems is Dave Nadler's freely available implementation.

Reply:
Maybe it will be useful if I will make a FREERTOS_MEMMANG variable optional. if it is defined, it will be used as above, otherwise the user will have to take care of including the equivalent of the files with pvPortMalloc and vPortFree implementations. The main idea was to keep the script as little as possible and give it as much flexibility as possible.